### DISCUSSION TITLE
Send later column makes thunderbird be really slow

### DISCUSSION THREAD

**Discussion by sukiletxe** [Extended-Thunder/send-later#400] [General]
So, I noticed that Thunderbird was really slow just when editing drafts or viewing the drafts folder (and focusing messages with arrow keys). I followed most advice in the [relevant guide section](https://extended-thunder.github.io/send-later/#thunderbird-hangs-frequently-on-windows) (great guide, by the way!), and I observed no noticeable difference. But when I disabled the send later column everything went smooth again. Now I have to right click the draft, click on open message in new tab, view source (ctrl+u), and search for `X-Send-Later` to view the scheduled date.

Just a solution to be added to the docs, in case it works for anybody else.

**Comment by jph-sendlater** [comment]
That's strange behavior. Hopefully once Thunderbird implements a more integrated solution for extensions to create columns this will no longer be the case.

In the mean time, there are several ways to check on your scheduled messages without opening the message source:

1. When you view the message in drafts, there should also be a header row (next to the 'Subject' line) which displays the schedule (this requires the option for "show send later header" be enabled in settings)
2. There is also an action button (next to the 'reply' and 'forward' buttons) which displays the schedule in a little popup dialog.

![Screenshot from 2021-11-12 11-35-00](https://user-images.githubusercontent.com/75693008/141525117-b962281a-5667-43ed-902b-5f457ab76cdf.png)

**Comment by sukiletxe** [comment]
Thanks for these tips! For some reason the last header appears as unknown for my screen reader (NVDA) when I tab, and I have to use object navigation to get to it. Similarly with the toolbar, though I can't reach that with tab nor F6. Luckily, the object navigation workaround exists.

As I said I hope this solution of not showing message columns gets written too in the user guide (I can do it too, if you wish).

**Comment by cloverdavid** [comment]
The column settings in Send Later do not stick. I have to change them every time I use Send Later. Suggestions?

**Comment by jikamens** [comment]
Closing because the old implementation of the Send Later column is gone in TB115 and will have to be redone from scratch when the Thunderbird add-on API maintainers get around to adding a proper API for add-ons to add custom columns.

One final thing to mention: if you click on the Send Later toolbar button in the main TB window it displays a list of scheduled messages.
